public/Leo Jayachandran/Glossarizer/libretextsglossarizer.js

```diff
@@ -0,0 +1,629 @@
+let LibreTextsGlossarizer = {
+    makeGlossary: async function (sourceOption) {
+        let glossaryRetriever = {
+            "textbook": async function getTextbookGlossaryPage() {
+                const coverPage = await LibreTexts.getCoverpage();
+                const subdomain = window.location.origin.split('/')[2].split('.')[0];
+                let glossaryPage = await LibreTexts.getAPI(`https://${subdomain}.libretexts.org/${coverPage}/zz%3A_Back_Matter/20%3A_Glossary`);
+                let data = await LibreTexts.authenticatedFetch(glossaryPage.id, 'contents?dream.out.format=json').then(response => {
```

Comment:
Good usage of both async/await as well as my custom functions! That reminds me that I need to improve documentation for reuse.js

Reply:
Do you really want to have both an await and then() tied to this? While I think I understand what you are doing I have never seen this dual usage of promise and await in the same line before so I am not familiar with its function.

Reply:
I decided to use the .then() function to avoid having to deal with the initial promise that hasn't been jsonified. It technically does the same thing as just using awaits to deal with it, at least if I understand it correctly, so I thought I'd keep it simpler, rather than add a new variable.

Reply:
As long as you've tested its functionality and that is works as you expect, it should be fine. The pattern I personally use is:
```suggestion
                let data = await LibreTexts.authenticatedFetch(glossaryPage.id, 'contents?dream.out.format=json');
                data = data.json(); //is await required here???
```
but I'll leave the decision up to you on what works for your purposes.

Reply:
So far, it's been working fine and can handle errors too, so I think I'm happy with the way it is right now